src/containers/Button.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+// component.
+import ButtonComponent from '../components/Button';
+
+class Button extends Component {
+  render() {
+    const { label, isIncrement } = this.props;
+    if (isIncrement) {
+      return <ButtonComponent label={label} action={this.props.onIncrement}/>;
+    }
+    return <ButtonComponent label={label} action={this.props.onDecrement}/>;
+  }
+}
+
+// Each container needs to have acess to the updated state. Hence, we need to subscribe to the store to listen to the changes.
+const mapStateToProps = (state) => {
+  console.log(state);
+  return {
+    number: state.number,
+  };
+};
+
+// Each container needs to dispatch events to the store. This is done by mapping to props.
+const mapDispatchToProps = (dispatch) => {
+  console.log(dispatch);
+  return {
```

Comment:
Although this should work, we are going to want to use "actions" file so that we can ensure to "flow type" things out.  

Example:
```
  import { counterChange } from '../llib/actions'

  // By using direction you can use the redux to add/subtract 1 within the state change. No need to pass value.
  ...return ({ onCounterChange: (direction: string) => dispatch(counterChange(direction) } ) 
```

actions.js
```
  export const counterChange = (direction: string) => ({ type: 'CHANGE_COUNTER', direction });
```

reducer
```
  ...
  switch (action.type) { 
    case : 'CHANGE_COUNTER': { 
      const { direction } = action
      let newCount = state.count;
      newCount = direction === 'increment' ? newCount++ : newCount--;
     return {
        count: newCount, 
     }
    } 
    .....
  }
```